### PR TITLE
celo-fullnode autoscaling

### PR DESCRIPTION
**Pull request by jcortejoso**
### Description

Add Horizontal Pod Autoscaler V2 resource to celo-fullnode statefulset.

### Other changes

- Included values file for the new celo-fullnode deployment which target is being used by blockscout indexer.
- Removed empty new lines

### Tested

Tested deploying and with templates.

### Related issues

- https://github.com/celo-org/infrastructure/issues/409
- https://github.com/celo-org/infrastructure/issues/326

### Backwards compatibility

_Brief explanation of why these changes are/are not backwards compatible._

### Documentation

_The set of community facing docs that have been added/modified because of this change_

### CODE REVIEW

File: packages/helm-charts/celo-fullnode/rc1-blockscout-archive-nodes.yaml [hunk 1]
```diff
@@ -0,0 +1,148 @@
+aws: false
+azure: false
+fullnameOverride: false
```
Comment (lvpeschke) [inline packages/helm-charts/celo-fullnode/rc1-blockscout-archive-nodes.yaml]: nit: these are the default values

File: packages/helm-charts/celo-fullnode/rc1-blockscout-archive-nodes.yaml [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  stack: blockscout
+  mode: archive
+gcp: true
+genesis:
+  epoch_size: 17280
+  genesisFileBase64: ""
+  network: rc1
+  networkId: 42220
+  useGenesisFileBase64: false
+geth:
```
Comment (lvpeschke) [inline packages/helm-charts/celo-fullnode/rc1-blockscout-archive-nodes.yaml:11]: does it make sense to add default values to the values.yaml file, where they are missing?

File: packages/helm-charts/celo-fullnode/rc1-blockscout-archive-nodes.yaml [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+metrics: true
+namespace: blockscout
+nodeSelector:
+  pool: t2d-standard-8-spot
+pprof:
+  enabled: true
+  path: /debug/metrics/prometheus
+  port: 6060
+prometheus: true
+replicaCount: 1
```
Comment (lvpeschke) [inline packages/helm-charts/celo-fullnode/rc1-blockscout-archive-nodes.yaml]: should this initially be higher?
or is the intention to increase it once the other archive node pool is scaled down?

**Review by lvpeschke — APPROVED** [review]

File: packages/helm-charts/celo-fullnode/rc1-blockscout-archive-nodes.yaml (hunk 2 above)
Comment (jcortejoso) [reply packages/helm-charts/celo-fullnode/rc1-blockscout-archive-nodes.yaml:11]: Yes, I plan to organize the values file with https://github.com/celo-org/infrastructure/issues/409

File: packages/helm-charts/celo-fullnode/rc1-blockscout-archive-nodes.yaml (hunk 3 above)
Comment (jcortejoso) [reply packages/helm-charts/celo-fullnode/rc1-blockscout-archive-nodes.yaml]: Once it is used we may have a different initial count, although this value will be affected by max and min replicas from hpa.

**Review by alvarof2 — APPROVED** [review]

File: packages/helm-charts/celo-fullnode/rc1-blockscout-archive-nodes.yaml (hunk 1 above)
Comment (jcortejoso) [reply packages/helm-charts/celo-fullnode/rc1-blockscout-archive-nodes.yaml]: Agree, better to just override non-default values.

Comment (jcortejoso) [reply packages/helm-charts/celo-fullnode/rc1-blockscout-archive-nodes.yaml]: Updated just setting non-default or particularly interesting values here.

**Pull request merged**